### DISCUSSION TITLE
Livewire wizard refreshes page automatically, and throws you back to first step.

### DISCUSSION THREAD

**Discussion by zerdotre** [spatie/laravel-livewire-wizard#74] [Q&A] [answered]
When I load my project, it loads normal. When proceeding to next step, it flashes the next step and then refreshes the component/page and throws you back to the first step.

Also is there any way to remember the state in session and continue the step we left onRefresh?

**Comment by freekmurze** [comment, marked as answer]
No, that is out of scope for the package. You should handle this in your own app.

**Reply by zerdotre** [reply]
Could you point me in the right direction?

**Reply by zerdotre** [reply]
After trying to override the state() method of stepcomponent and using setAllState() etc. I found that it was rather easily solved by
1: `$request->session()->put('state', $this->state()->all());` in the mount() method of all stepcomponents.
2: checking for session('state') in controller, and including livewire component with 'show-step="" :initialState=""'.
